app/marketing/views.py

```diff
@@ -885,12 +886,42 @@ def trending_quests():
         ).order_by('-recent_attempts').all()[0:10]
     return quests
 
+def quest_of_the_day():
+    quest = trending_quests()[0]
+    return quest
+
+def upcoming_grant():
+    grant = Grant.objects.order_by('-weighted_shuffle').first()
+    return grant
+
+def upcoming_hackathon():
+    try:
+        return HackathonEvent.objects.filter(end_date__gt=timezone.now()).order_by('-start_date')
+    except HackathonEvent.DoesNotExist:
+        try:
+            return HackathonEvent.objects.filter(start_date__gte=timezone.now()).order_by('start_date').first()
+        except HackathonEvent.DoesNotExist:
+            return None
+
+def latest_activities(user):
+    from retail.views import get_specific_activities
+    cutoff_date = timezone.now() - timezone.timedelta(days=7)
+    # activities = Activity.objects.filter(created_on__gte=cutoff_date).order_by('-created_on')[:4]
```

Comment:
leftover code

Reply:
Done in [3e561b6](https://github.com/gitcoinco/web/pull/6471/commits/3e561b647f26c969686f74c28a11de7b971b2d91)